Address feedback from #40032 (#40049)
dotnet/docs: docs/core/deploying/index.md

Cross-platform binaries can be run on any operating system as long as the targeted .NET runtime is already installed. If the targeted .NET runtime isn't installed, the app may run using a newer runtime if the app is configured to roll-forward. For more information, see [framework-dependent apps roll forward](../versions/selection.md#framework-dependent-apps-roll-forward).

> You can choose to run the app directly with a platform-specific binary as the app host or with the `dotnet` as the host. There should be no app behavior difference when launching from via app host versus the `dotnet` host for ordinary server apps. Launching via an app host gives you better integration with the underlying OS. For example:

Reviewer: ```suggestion
You can choose to run the app as a platform-specific executable or as a cross-platform binary via `dotnet` command. There should be no app behavior difference when launching the platform-specific executable versus the `dotnet` command for ordinary server apps Launching via a platform-specific executable gives you better integration with the underlying OS. For example:
```
The terms used in the rest of this document are "platform-specific executable" and "cross-platform binary". "Host" is not used in the rest of the document. Should we be consistent?

- You see the app host name in your process list and not `dotnet`, which could be confusing if there's more than one.

> - You can customize the app host with OS specific features. For example, see [this discussion about configuring default stack size on Windows](https://github.com/dotnet/runtime/issues/96347#issuecomment-1981470713).

Reviewer: ```suggestion
- You can customize the platform-specific executable with OS specific features. For example, see [this discussion about configuring default stack size on Windows](https://github.com/dotnet/runtime/issues/96347#issuecomment-1981470713).
```
